airbyte-integrations/connectors/destination-chroma/destination_chroma/destination.py

```diff
@@ -0,0 +1,83 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+from typing import Any, Iterable, Mapping
+
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.destinations import Destination
+from airbyte_cdk.destinations.vector_db_based.embedder import CohereEmbedder, Embedder, FakeEmbedder, OpenAIEmbedder
+from airbyte_cdk.destinations.vector_db_based.indexer import Indexer
+from airbyte_cdk.destinations.vector_db_based.writer import Writer
+from airbyte_cdk.models import AirbyteConnectionStatus, AirbyteMessage, ConnectorSpecification, ConfiguredAirbyteCatalog, DestinationSyncMode, Status
+
+from destination_chroma.config import ConfigModel
+from destination_chroma.indexer import ChromaIndexer
+
+BATCH_SIZE = 128
+
+embedder_map = {"openai": OpenAIEmbedder, "cohere": CohereEmbedder, "fake": FakeEmbedder}
+
+class DestinationChroma(Destination):
+
+    indexer: Indexer
+    embedder: Embedder
+
+    def _init_indexer(self, config: ConfigModel):
+        self.embedder = embedder_map[config.embedding.mode](config.embedding)
+        self.indexer = ChromaIndexer(config.indexing, self.embedder)
+
+    def write(
+        self, config: Mapping[str, Any], configured_catalog: ConfiguredAirbyteCatalog, input_messages: Iterable[AirbyteMessage]
+    ) -> Iterable[AirbyteMessage]:
+
+        """
+        Reads the input stream of messages, config, and catalog to write data to the destination.
+
+        This method returns an iterable (typically a generator of AirbyteMessages via yield) containing state messages received
+        in the input message stream. Outputting a state message means that every AirbyteRecordMessage which came before it has been
+        successfully persisted to the destination. This is used to ensure fault tolerance in the case that a sync fails before fully completing,
+        then the source is given the last state message output from this method as the starting point of the next sync.
+
+        :param config: dict of JSON configuration matching the configuration declared in spec.json
+        :param configured_catalog: The Configured Catalog describing the schema of the data being received and how it should be persisted in the
+                                    destination
+        :param input_messages: The stream of input messages received from the source
+        :return: Iterable of AirbyteStateMessages wrapped in AirbyteMessage structs
+        """
+
+        config_model = ConfigModel.parse_obj(config)
+        self._init_indexer(config_model)
+        writer = Writer(config_model.processing, self.indexer, batch_size=BATCH_SIZE)
```

Comment:
The most recent version of the CDK changes this interface a bit - now the embedder is not passed to the indexer, but to the writer. Please check the PRs for Milvus and Weaviate: https://github.com/airbytehq/airbyte/pull/29790/files#diff-70cdc7d8dd268c2d1bd741d5b10c4484ee17a1369133cf9586336979285b8839 https://github.com/airbytehq/airbyte/pull/30151/files#diff-2e98784f6ca6efcc82e7c3494eb6a60a4c99b68ba807df7582c87a8017b609ec